docs/content/tutorial/connecting-to-external-services.mdx

```diff
@@ -0,0 +1,186 @@
+description: Learn how to connect to external services in Dagster with resources
+---
+
+Throughout the tutorial, you have built a data pipeline that connects to different external services, such as Hacker News. By building this asset graph, Dagster acts as a control plane of multiple different external services. However, connecting to these services and managing their use can be difficult. Resources are the recommended way to manage these connections in Dagster.
+
+Resources are Dagster objects that connect to external services, such as Slack, Snowflake, or S3. In this section, you’ll learn more about resources, including how to:
+
+- Define a resource and use it in an asset
+- Configure a resource
+- Customize a resource based on the environment where Dagster is running
+
+## Why use resources?
+
+In the `topstory_ids` and `topstories` assets, you used the `requests` library to communicate with the Hacker News API. This code gets harder to maintain if:
+
+- Connections need to be shared across multiple assets
+- You need to authenticate with the external service
+- Configurations need to be consistent across its usage
+- If any of the above changes based on environment (e.g. development vs. production)
+
+Aside from the complexity of configuring them, building data pipeline creates many connections to external services. It becomes difficult to understand what assets use which connections or how often a connection is used.
+
+You should use resources to manage communicating with external services because they:
+
+- Enable flexibility in configuration
+- Standardize how connections are used across assets
+- Create a single source to monitor connections in the Dagster UI
+
+---
+
+# Step 1: Setting up a resource
+
+Suppose your Hacker News pipeline in Dagster has picked up momentum, and your stakeholders want to learn more about who is using it. You are tasked with analyzing how many people sign up for Hacker News.
+
+In the scaffolded Dagster project you made in Part 2 of the tutorial, you'll have seen a directory called `resources` with some files in it. The most important file is the `resources.py` file that exposes a resource called `DataGeneratorResource`. This resource generates simulated data about Hacker News signups. You'll use this resource to get the data needed to produce an asset for analysis.
+
+<Note>
+  The signup data from Hacker News is fake and generated by the library. This is
+  simulated data and should not be used for real use cases.
+</Note>
+
+In your `__init__.py`, import the class, create an instance of it, and add it to the `resources` argument for your code location's `Definitions` object under the key `hackernews_api`. The key used to define a resource in the `Definitions` object is the key you'll use to reference the resource later in your code. In this case, we’ll call it `hackernews_api`.
+
+Verify that your code looks similar to the code below:
+
+```python
```

Comment:
do we plan to check in those code snippets somewhere so we can keep stuff tested and up-to-date?
maybe alongside with https://github.com/dagster-io/dagster/pull/14560 (and maybe name it as `tutorial_template`?), we have a checked-in complete version called `tutorial_completed"?

Reply:
🖖🏽 Yeup! The plan is to check in and test the snippets once we can agree on content, in case anything gets checked

Also great idea with putting a `tutorial_completed` template!